{
    "source": "BDSA",
    "name": "BDSA-2020-2486",
    "title": "Debian RDFLib Vulnerable to Remote Code Execution (RCE) via Improper Loading of Modules",
    "description": "Debian RDFLib is vulnerable to remote code execution (RCE) due to improper loading of modules. An attacker could exploit this vulnerability by supplying a system with maliciously crafted input.\n\n**Note:** This issue is specific to use of the debian/scripts directory.",
    "technicalDescription": "The `python-rdflib-tools` package for RDFLib has CLI tools which can load Python Modules from the current working directory. This occurs because `python -m` looks in the current working directly. \n\nThe issue was mitigated by implementing the use of Python-provided wrapper scripts.",
    "publishedDate": "2020-10-13T13:23:50.906Z",
    "updatedDate": "2020-10-13T13:23:50.894Z",
    "disclosureDate": "2020-09-23T00:00:00.000Z",
    "exploitPublishDate": "2019-02-08T00:00:00.000Z",
    "solution": "Fixed in [**4.2.2-2**](https://salsa.debian.org/python-team/packages/rdflib/-/tags/debian%2F4.2.2-2) by [this](https://salsa.debian.org/python-team/packages/rdflib/-/commit/ecde5712341b74b483d140f6e89b1f274771602f) commit.",
    "severity": "HIGH",
    "cvss2": {
        "baseScore": 7.5,
        "impactSubscore": 6.4,
        "exploitabilitySubscore": 10.0,
        "severity": "MEDIUM",
        "accessVector": "NETWORK",
        "accessComplexity": "LOW",
        "authentication": "NONE",
        "confidentialityImpact": "PARTIAL",
        "integrityImpact": "PARTIAL",
        "availabilityImpact": "PARTIAL",
        "temporalMetrics": {
            "exploitability": "PROOF_OF_CONCEPT",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 5.9
        },
        "vector": "(AV:N/AC:L/Au:N/C:P/I:P/A:P/E:POC/RL:OF/RC:C)"
    },
    "cvss3": {
        "baseScore": 9.8,
        "impactSubscore": 5.9,
        "exploitabilitySubscore": 3.9,
        "severity": "HIGH",
        "attackVector": "NETWORK",
        "attackComplexity": "LOW",
        "confidentialityImpact": "HIGH",
        "integrityImpact": "HIGH",
        "availabilityImpact": "HIGH",
        "privilegesRequired": "NONE",
        "scope": "UNCHANGED",
        "userInteraction": "NONE",
        "temporalMetrics": {
            "exploitability": "PROOF_OF_CONCEPT",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 8.8
        },
        "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H/E:P/RL:O/RC:C"
    },
    "useCvss3": true,
    "classifications": [],
    "zeroDay": false,
    "underReview": false,
    "parentAdvisory": false,
    "credit": "Gabriel Corona",
    "workaround": "",
    "vendorFixDate": "2019-02-14T00:00:00.000Z",
    "vendorNotifiedDate": "2019-02-08T00:00:00.000Z",
    "bdsaTags": [
        "RCE"
    ],
    "overallScore": 8.8,
    "_meta": {
        "allow": [
            "GET"
        ],
        "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2020-2486",
        "links": [
            {
                "rel": "cwes",
                "href": "https://trendmicro2.app.blackduck.com/api/cwes/CWE-94"
            },
            {
                "rel": "related-vulnerability",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/CVE-2019-7653",
                "label": "NVD"
            },
            {
                "rel": "bdsa-ranges",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2020-2486/ranges"
            },
            {
                "rel": "reference",
                "href": "https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921751",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921751",
                "label": "POC"
            },
            {
                "rel": "reference",
                "href": "https://lists.debian.org/debian-lts-announce/2019/03/msg00019.html",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://salsa.debian.org/python-team/packages/rdflib/-/commit/ecde5712341b74b483d140f6e89b1f274771602f",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://salsa.debian.org/python-team/packages/rdflib/-/tags/debian%2F4.2.2-2",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "default-remediation-status",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2020-2486/default-remediation-status"
            }
        ]
    }
}